### PR TITLE
Issue #590: rename log_dir to exp_name and log_root to log_dir

### LINKED ISSUE #590: Clarify log_root and log_dir for train/predict
## Subject of the issue

It is quite confusing which is which and what's their purpose.

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

- Rename `log_dir` to `exp_name`
- Rename `log_root` to `log_dir`

Fixes #590

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [x] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mathpluscode** [comment]
So I did

- Rename log_dir to exp_name
- Rename log_root to log_dir

It might be a bit confusing, but now

- log_dir tells you where all logs are saved
- exp_name let you identify each experiment/run

I've updated the docs correspondingly.

@YipengHu @NMontanaBrown @s-sd would you have time to review this?

**Review by YipengHu — CHANGES_REQUESTED** [review]
so need to clarify this.

	if experiment/run name is not provided, each will be saved in to a time-stamped folder
	if provided, the experiment/run name will be the sub-folder name under the log_dir, without time-stamp
	if log_dir is not provided, *** will be used as default...

**Comment by mathpluscode** [comment]
image
we have this docs

**Review by YipengHu — COMMENTED** [review]
- **Log directory**:
  `--log_dir`, specifies the log directory for logging output information and results.
  By default, it is `logs` under the package root.
  Example usage:
  - `--log_dir logs` for specifying the log directory `logs/` under current directory.
- **Experiment name**:
  `--exp_name` or `-n`, specifies the name of an experiment (every time a training or a prediction is run),
  which will be used together with the log directory (via `log_dir`) to specify the sub-folder that saves 
  the output information and results from individual experiments (runs).
  If this is not provided, it creates a timestamp-named sub-folder under the `log_dir`, by default, e.g.
  `logs/20200810-194042/`.
  Example usage:
  - `--exp_name test --log_dir logs` for saving under `logs/test/`.  
  - `--log_dir logs` for saving under `logs/20210101-120000/`. # assuming `20210101-120000` is current time
  - `--exp_name test` for saving under `~/DeepReg/logs/test/`. # assuming `~/DeepReg` is the package root.

**Comment by YipengHu** [comment]
does log_dir has no shorthand '-l' now?

File: deepreg/predict.py [hunk 1]
```diff
@@ -148,21 +148,21 @@ def predict_on_dataset(
...
     """
     Function to create new directory to log directory to store results.
 
-    :param config_path: string or list of strings, path of configuration files
-    :param log_root: str, root of logs
-    :param log_dir: string, path to store logs.
-    :param ckpt_path: str, path where model is stored.
+    :param config_path: path of configuration files.
+    :param log_dir: path of the directory.
+    :param exp_name: name of the experiment.
```
Comment (YipengHu) [inline deepreg/predict.py:158]: use searchable "log directory" and "experiment name"

**Review by YipengHu — CHANGES_REQUESTED** [review]
it turned out to be mainly a doc problem really. i have no further comments.

File: deepreg/predict.py (hunk 1 above)
Comment (mathpluscode) [reply deepreg/predict.py]: updated

**Review by YipengHu — APPROVED** [review]

**Review by s-sd — APPROVED** [review]

**Pull request merged**